pytests/common.py

```diff
@@ -161,13 +161,19 @@ def test_func():
                     with TimeLimit(10, env, "Failed waiting for replica to be in sync"):
                         while True:
                             replication_info = con.execute_command('info', 'replication')
-                            if replication_info['slave0']['state'] == 'online':
-                                break
+                            try:
+                                if replication_info['slave0']['state'] == 'online':
+                                    break
+                            except KeyError:
+                                pass
                             time.sleep(0.1)
             version = env.cmd('info', 'server')['redis_version']
             if skipOnRedis6 and '6.0' in version:
                 env.skip()
             if test_function.__doc__ is not None:
+                if withReplicas:
+                    num_replicas = shardsCount - 1
+                    env.expect('WAIT', num_replicas, 0).equal(num_replicas if decodeResponses else num_replicas.to_bytes())
```

Comment:
The wait should come after the `function load` and we should do the wait on each shard like we do here:
https://github.com/RedisGears/RedisGears/pull/863/files#diff-1cf27a9d4d71117ed40ffa9de5085d314d7782a78e8c8620fdd4f6c7cf578511R160

Reply:
Notice that the number of shards should not be used on the wait command, we always have at most one replica per shard.